OSDOCS-3068: Adding an intro to Updates docs (#41268)
openshift/openshift-docs: updating/understanding-openshift-updates.adoc

:_content-type: ASSEMBLY
[id="understanding-openshift-updates"]
= Understanding {product-title} updates
include::modules/common-attributes.adoc[]
:context: understanding-openshift-updates

toc::[]

> With {product-title} 4, you can update a {product-title} cluster with a single operation by using the web console or the OpenShift CLI (`oc`). Platform administrators are automatically notified when an update is available for their cluster. Updates can be applied either from the `oc` command-line interface (CLI) or the web console.

Reviewer: I think we can drop the final sentence here, now that the first sentence covers the UI choices.

The OpenShift Update Service (OSUS) builds a graph of upgrade possibilities based on release images in the registry.

> The graph is based on supported, tested update paths from a specific version. {product-title} clusters connect to the Red Hat Hybrid Cloud servers and identify which clusters the user is running, along with the version information. OSUS responds with information about known update targets. Either a cluster administrator or an automatic update controller edits the custom resource (CR) of the Cluster Version Operator (CVO) with the new version for OSUS to update to.

Reviewer: `candidate-4.y` channels include releases and updates that are not supported.  Can we say "...recommended, tested..." or some such, to be more generic?  And leave supported-ness up to the channel semantics?

Reviewer: another nit: `with the new version for OSUS to update to.`  It's actually the CVO making the update.  The flow is:

1. CVO polls OSUS for possible targets.
2. CVO stores OSUS response in ClusterVersion.
3. Admin/bot selects a possible target from the ClusterVersion `status` (or brings in their own explicit pullspec, but probably not worth involving this path here), and updates ClusterVersion `spec.desiredUpdate`.
4. CVO sees the re-target request in `spec.desiredUpdate`, performs some "do I really want to do this?" validation, and then begins updating.

So OSUS is out of the picture well before step 4's "we're starting to update".

:_content-type: ASSEMBLY
[id="understanding-openshift-updates"]
= Understanding {product-title} updates
include::modules/common-attributes.adoc[]
:context: understanding-openshift-updates

toc::[]

> With {product-title} 4, you can update a {product-title} cluster with a single click using over the air update.

Reviewer: I'm not a big fan of "over the air", because things work pretty much the same way in restricted networks (although it may take a few setup steps to get the target bits somewhere the cluster can reach them).  Maybe "... with a single operation." or some such?

> With {product-title} 4, you can update a {product-title} cluster with a single operation using the web console or the OpenShift CLI (`oc`).
> Platform administrators are automatically notified when an update is available for their cluster.
> Updates can be applied either from the `oc` command-line interface (CLI) or the web console.

Reviewer 2: Small nits/suggestions:
```suggestion
With {product-title} 4, you can update a {product-title} cluster with a single operation by using the web console or the OpenShift CLI (`oc`). Platform administrators are automatically notified when an update is available for their cluster. Updates can be applied either from the `oc` command-line interface (CLI) or the web console.
```

> The {product-title} update service builds a graph of upgrade possibilities based on release images in the registry.

Reviewer 2: Should this be "the OpenShift Update Service (OSUS)"? And I would remove the hard wraps here as well.

```suggestion
The OpenShift Update Service (OSUS) builds a graph of upgrade possibilities based on release images in the registry.
```

Also, if you end up making ^^this change, then you probably don't need to spell out OSUS under the CVO definition. I hope that makes sense. 😛

> This graph is based on allowable tested update paths from a specific version.

Reviewer: "allowable" feel controlling.  Cluster administrators are able to make off-graph updates, and sometimes those are still supported via support exceptions.  Can we say `supported, tested update paths` or some such here, or do we need to find a word that doesn't tie so directly to contracts?

{product-title} clusters connect to the Red Hat Hybrid Cloud servers and identify which clusters the user is running along with version information.

> Policy engines combine the information from customer entitlement in conjunction with the update graph to tell the cluster which release they can upgrade to.

Reviewer: I think "policy engines" is getting into internal implementation details for the update service.  Those are useful details when we're talking to folks who are about to run their own local update service.  But this doc seems to be aimed more at the admin thinking about updating, in which case the detail may be distracting.  Maybe `The OpenShift Update Service responds with information about known update targets.` or some such, because now that we have conditional updates it's broader than `...about recommended updates`, and if someone is using a `candidate-4.*` channel, it's broader than `...about supported updates`.

> Either a cluster administrator or through the automatic update controller edits the Cluster Version Operator (CVO)’s customer resource (CR) with the new version to update to.

Reviewer: `Either a cluster administrator or an automatic update controller edits ...`?  Or is there a specific automatic update controller we're talking about here?

Reviewer: also "CR" is "[custom resource][1]", not "customer resource".

[1]: https://www.bmc.com/blogs/kubernetes-crd-custom-resource-definitions/

> The graph is based on supported, tested update paths from a specific version.
> {product-title} clusters connect to the Red Hat Hybrid Cloud servers and identify which clusters the user is running along with version information.
> The OpenShift Update Service responds with information about known update targets.
> Either a cluster administrator or an automatic update controller edits the Cluster Version Operator (CVO)’s custom resource (CR) with the new version to update to.
> Once the CVO receives the update image from the registry, the CVO then applies the changes.

Reviewer 2: A few suggestions that might help clarify, or might not: 😄 
```suggestion
The graph is based on supported, tested update paths from a specific version. {product-title} clusters connect to the Red Hat Hybrid Cloud servers and identify which clusters the user is running, along with the version information. OSUS responds with information about known update targets. Either a cluster administrator or an automatic update controller edits the custom resource (CR) of the Cluster Version Operator (CVO) with the new version for OSUS to update to.
After the CVO receives the update image from the registry, the CVO then applies the changes.
```
